src/Client.php

```diff
@@ -634,6 +670,14 @@ protected function call($method, $uri, $params = null)
                 $this->checkThrottling($response);
 
                 $continue = false;
+            } catch (AuthorizationException $e) {
+                if ($this->getLoginFlag()) {
+                    throw $e;
+                }
+
+                $this->refreshLogin();
+                $this->call($method, $uri, $params);
```

Comment:
For some reason if I don't break from the current loop, it returns Unauthorized responses for calls after refreshLogin(). So what I did is to call same function recursevily and break from the current loop.
Also I have seen that by calling login() instead of refreshLogin() there is no such problem.
But I can't find the problem why is it like that.

So a possibility would be to call login() instead of refreshLogin(). Because the response for both API calls is the same.

Reply:
I guess that issue appears because we refresh the tokens but don't update them for the current request (since they are assigned outside the loop). Maybe that's what caused the infinite loop all along.

Please check if following approach fixes the issue:
* Call `Client::setDefaultParams` instead of `Client::call`.
* Drop the `break`.